crates/server/src/store/sqlite/mod.rs

```diff
@@ -752,7 +752,18 @@ impl SqliteAuthority {
 
                     // zone     rrset    rr       Add to an RRset
                     info!("upserting record: {:?}", rr);
-                    updated = self.in_memory.upsert(rr.clone(), serial).await || updated;
+                    let upserted = self.in_memory.upsert(rr.clone(), serial).await;
+
+                    #[cfg(all(feature = "metrics", feature = "__dnssec"))]
+                    if auto_signing_and_increment {
+                        if upserted {
+                            self.metrics.persistent.dynamically_add();
```

Comment:
Are you sure `upserted` represents addition?

Reply:
I think I had checked that with a quick test. As well the doc comment on the `upsert()` method states that `true will only be returned on insert.
https://github.com/hickory-dns/hickory-dns/blob/11e8639a2ff7c228ddc4ec5e3d1c87362b73d4f5/crates/server/src/store/in_memory/mod.rs#L199-L211